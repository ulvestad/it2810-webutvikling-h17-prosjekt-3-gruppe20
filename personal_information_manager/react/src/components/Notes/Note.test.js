import React from 'react';
import sinon from 'sinon';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { expect } from 'chai';

import Note from './Note';

const jestExpect = global.expect;

it('renders without crashing', () => {
  shallow(<Note />);
});

it('renders correctly without props', () => {
  const tree = renderer.create(<Note />).toJSON();
  jestExpect(tree).toMatchSnapshot();
});

it('renders correctly with props', () => {
  const
      id = 1234,
      title = 'Title',
      bodyContent = 'This is the body',
      handleDelete = () => {console.log('deleted')};

  const noteComponent =
      <Note
          id={id}
          title={title}
          body={bodyContent}
          handleDelete={handleDelete}
      />;

  const tree = renderer.create(noteComponent).toJSON();

  jestExpect(tree).toMatchSnapshot();
});

it('calls handler when clicked', () => {
  const handler = sinon.spy();
  const wrapper = shallow(<Note handleDelete={handler} id={2} />);

  wrapper.find('button').simulate('click');
  expect(handler).to.have.property('callCount', 1);
});

it('calls handler with correct id', () => {
  const handler = sinon.spy();
  const wrapper = shallow(<Note handleDelete={handler} id={2} />);

  wrapper.find('button').simulate('click');
  expect(handler.args[0][0]).to.equal(2);
});

// React-modal uses portals, which are problematic at the current time
// Enzyme does not funny support them, and mount fails
// https://github.com/airbnb/enzyme/issues/1150
