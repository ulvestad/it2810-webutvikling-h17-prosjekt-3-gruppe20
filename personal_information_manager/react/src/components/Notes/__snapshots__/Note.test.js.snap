// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with props 1`] = `
<div
  className="panel panel-default note"
>
  <div
    className="panel-heading note-heading"
  >
    <button
      aria-label="Close"
      className="close"
      onClick={[Function]}
      type="button"
    >
      <span
        aria-hidden="true"
      >
        ×
      </span>
    </button>
    <h3
      className="panel-title"
    >
      Title
    </h3>
  </div>
  <div
    className="panel-body"
  >
    <p>
      This is the body
    </p>
  </div>
</div>
`;

exports[`renders correctly without props 1`] = `
<div
  className="panel panel-default note"
>
  <div
    className="panel-heading note-heading"
  >
    <button
      aria-label="Close"
      className="close"
      onClick={[Function]}
      type="button"
    >
      <span
        aria-hidden="true"
      >
        ×
      </span>
    </button>
    <h3
      className="panel-title"
    />
  </div>
  <div
    className="panel-body"
  >
    <p />
  </div>
</div>
`;
